### PR TITLE
chore: update typescript to v4.4.0

### PR DESCRIPTION
**Pull request by bryceosterhaus**
fixes https://github.com/liferay/liferay-frontend-projects/issues/627

Tried this locally and didn't see errors, should be good to go?

### CODE REVIEW

File: projects/npm-tools/packages/npm-scripts/package.json
Comment (jbalsas) [inline projects/npm-tools/packages/npm-scripts/package.json:79]: Should we prefer a fixed version `4.4.2` here or at least `~4.4.2`? After all, `^4.2.3` also works for `4.4.2` ;) 

Based on the experience here, I get the feeling we want to be very explicit about which specific version we're using (and shipping). That way, devs in Liferay DXP might have an easier time figuring out which TS options are available to them.

This makes me think that we might also want to start being more vocal about TS support so we get more people on board, but maybe that can wait until we land some bits of https://github.com/liferay-frontend/liferay-portal/pull/1379

**Review by jbalsas — APPROVED** [review]
LGTM based on our current setup, but based on the experience on trying to figure out which TS features are available (can't rely just on `package.json` broad version, really), I'd say it would be better if we restricted the version range a bit more.

File: projects/npm-tools/packages/npm-scripts/package.json
Comment (bryceosterhaus) [reply projects/npm-tools/packages/npm-scripts/package.json]: Totally agree. Just pushed an update

**Pull request merged**
